neutron/beamline.cpp

```diff
@@ -67,6 +67,17 @@ template <class T> static Variable two_theta(const T &d) {
 
   return acos(dot(beam, scattered));
 }
+
+template <class T> static VariableConstView incident_energy(const T &d) {
+  return d.meta().contains(Dim::IncidentEnergy) ? d.meta()[Dim::IncidentEnergy]
+                                                : VariableConstView{};
+}
+
+template <class T> static VariableConstView final_energy(const T &d) {
+  return d.meta().contains(Dim::FinalEnergy) ? d.meta()[Dim::FinalEnergy]
+                                             : VariableConstView{};
+}
+
```

Comment:
Minor point:  it might be useful to have a `is_elastic` or `is_inelastic` method here. This will prevent having to do double checks elsewhere for example checking you don't have both incident and final energies.

Reply:
Maybe at some point, right now it would make things messier, I think? If we have both Ei and Ef I suppose both `is_elastic` and `is_inelastic` would returns `false`? Then `incident_energy` as well as `final_energy` would still returns an empty variable, and we would still have to do the check in `tof_to_energy_transfer`.